src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/OffenderService.kt

```diff
@@ -97,4 +98,51 @@ OffenderService(
       }
     }
   }
+
+  @Transactional
+  fun updateOffenderDetails(nomsId: String, request: UpdateOffenderDetailsRequest) {
+    val licenceStatuses = listOf(
+      LicenceStatus.IN_PROGRESS,
+      LicenceStatus.SUBMITTED,
+      LicenceStatus.APPROVED,
+      LicenceStatus.VARIATION_IN_PROGRESS,
+      LicenceStatus.VARIATION_SUBMITTED,
+      LicenceStatus.VARIATION_APPROVED,
+      LicenceStatus.VARIATION_REJECTED,
+      LicenceStatus.ACTIVE,
+    )
+    val existingLicences = this.licenceRepository.findAllByNomsIdAndStatusCodeIn(nomsId, licenceStatuses)
+    var offenderDetailsChanged = false
```

Comment:
Mutating state in a `map` is a bit of a code smell... generally would try avoiding it in higher order functions.  

One way around this is just splitting up the operation which would break down the operation into declarative steps and lead to an overall simpler approach. 
```
  val licencesToChange = existingLicences.filter ...
  val updatedLicences = licencesToChange.map ...
  if (updatedLicences.isNotEmpty) {
     ...
```

Reply:
Agreed that while technically less performant, it will be unnoticeable.
I can't think of a scenario when someone would have more than two licences at once (now that `inactive` licences are excluded), so iterating over an array of 0-2 objects twice rather than once shouldn't cause the downfall of our app 😂